### ISSUE TITLE
Fix accuracy issues when printing result tables and redesign TableBuilders for more flexibility

### ISSUE THREAD

**Issue by StruffelProductions** [cloudsimplus/cloudsimplus#404]
# ISSUE

## Actual behavior 

The `roundTime()` function in CloudletsTableBuilder causes incorrect FinishTime display in MinTimeBetweenExample.java

Adjusting the simulation parameters such that VM_MIPS = 100 or CLOUDLET_LENGTH=1000 (basically anything that causes exec time to increase) causes the results table to only apply rounding in some fields. This is particularly noticeable because this example overwrites the default formatting in order to display more decimal digits:

```
|Cloudlet|Status |DC|Host|Host PEs |VM|VM PEs   |CloudletLen|FinishedLen|CloudletPEs|StartTime|FinishTime|ExecTime
|--------|-------|--|----|---------|--|---------|-----------|-----------|-----------|---------|----------|--------
|      ID|       |ID|  ID|CPU cores|ID|CPU cores|         MI|         MI|  CPU cores|  Seconds|   Seconds| Seconds
|       3|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,17|      2,00|    1,00
|       5|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,64|      1,00|    1,44
|       2|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,70|      1,00|    1,42
|       0|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,87|      1,00|    1,37
|       1|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,91|      1,00|    1,35
|       4|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,96|      1,00|    1,34
```

## Expected behavior

I would expect the table to show increased precision on all fields.

## If the issue is related to a specific method, provide a test case that fails in order to show the problem

I was already working in the CloudletsTableBuilder.java file for a future PR so I simply modified the function `roundTime()` to just `return time;`, effectively skipping it.
This resolved the issue in this particular example (see output below) but created an inaccuracy in VmDestructionExample (Start = 0, Finish = 15, ExecTime= 14). 

New MinTimeBetweenExample.java output (now fixed):
```
|Cloudlet|Status |DC|Host|Host PEs |VM|VM PEs   |CloudletLen|FinishedLen|CloudletPEs|StartTime|FinishTime|ExecTime
|--------|-------|--|----|---------|--|---------|-----------|-----------|-----------|---------|----------|--------
|      ID|       |ID|  ID|CPU cores|ID|CPU cores|         MI|         MI|  CPU cores|  Seconds|   Seconds| Seconds
|       3|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,17|      1,69|    1,52
|       5|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,64|      2,09|    1,44
|       2|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,70|      2,13|    1,42
|       0|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,87|      2,24|    1,37
|       1|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,91|      2,26|    1,35
|       4|SUCCESS| 1|   0|        4| 0|        4|       1000|       1000|          1|     0,96|      2,30|    1,34
```

New VmDestructionExample.java (now broken):
```
                                               SIMULATION RESULTS

|Cloudlet|Status |DC|Host|Host PEs |VM|VM PEs   |CloudletLen|FinishedLen|CloudletPEs|StartTime|FinishTime|ExecTime
|--------|-------|--|----|---------|--|---------|-----------|-----------|-----------|---------|----------|--------
|      ID|       |ID|  ID|CPU cores|ID|CPU cores|         MI|         MI|  CPU cores|  Seconds|   Seconds| Seconds
|       0|SUCCESS| 1|   0|        8| 0|        4|      10000|      10000|          2|        0|        15|      14
|       2|SUCCESS| 1|   0|        8| 0|        4|      10000|      10000|          2|        0|        25|      25
```

My personal suggestion would be to remove the rounding function and instead display one decimal point of precision for Start/Finish/ExecTime by default and give them an easy way to increase precision even further if needed. This way users see that the times are decimal values and that some rounding errors are to be expected. Otherwise I believe that variations of this rounding issue are going to crop up in other situations again and again.

A pull request for this proposal is in the works.

## Specifications like the version of the project, operating system or workload file used
Tested on:

cloudsimplus master 5b8877c
cloudsimplus-examples master aa4b38b

**Comment by manoelcampos** [comment]
> ## If the issue is related to a specific method, provide a test case that fails in order to show the problem
> I was already working in the CloudletsTableBuilder.java file for a future PR so I simply modified the function `roundTime()` to just `return time;`, effectively skipping it. This resolved the issue in this particular example (see output below) but created an inaccuracy in VmDestructionExample (Start = 0, Finish = 15, ExecTime= 14).

Great, but in that case, just remove the `roundTime()` function instead of making it return the exact value.
Keeping the function will just cause confusion, since it won't do what it's supposed to do.

**Comment by manoelcampos** [comment]
> New VmDestructionExample.java (now broken):
> 


If we include the decimal places back in that example, it will fix the precision issue.

**Comment by manoelcampos** [comment]
> My personal suggestion would be to remove the rounding function and instead display one decimal point of precision for Start/Finish/ExecTime by default and give them an easy way to increase precision even further if needed. 

That can be achieved by changing the constant in `CloudletsTableBuilder` to an instance field and adding a getter and setter.

**Comment by StruffelProductions** [comment]
> > My personal suggestion would be to remove the rounding function and instead display one decimal point of precision for Start/Finish/ExecTime by default and give them an easy way to increase precision even further if needed.
> 
> That can be achieved by changing the constant in `CloudletsTableBuilder` to an instance field and adding a getter and setter (with 1 decimal place as default).

Yes, although the architecture makes this a bit more complicated. Because `createTableColumns()` gets called during the constructor of the superclass which means that the (non-static) variables of the class are not yet initialized. To solve this I created a small initialization function which ensures that the default values are set at the right moment but there might be a more elegant solution that I am not yet seeing. Here is the current progress: https://github.com/cloudsimplus/cloudsimplus/compare/master...StruffelProductions:cloudsimplus:table-time-formatting

While I was at it I also added formatting support for the other fields for things like leading zeros or scientific notation of large numbers but I will explain that in the proper PR. Also planning to submit an example to the example-repo.
